### DISCUSSION TITLE
Hide delete action button

### DISCUSSION THREAD

**Discussion by eaziweb95** [filamentphp/filament#2695] [Help] [answered]
### Package

filament/filament

### Package Version

2

### Laravel Version

9.0

### Livewire Version

2

### PHP Version

8

### Bug description

I use this form becasuse i don't find any forums to add questions
I'd like to have an example to change the actions buttons in a table.
I need to hide the Delete action button when my record is inactive (flag actif) 
Thanks for your help

### Steps to reproduce

Hide delete action button

### Relevant log output

```shell
Hide delete action button
```

**Comment by danharrin** [comment, marked as answer]
We have GitHub Discussions and [Discord](https://filamentphp.com/discord) for questions...

`->hidden(fn ($record) => ! $record->is_active)`

**Reply by danharrin** [reply]
I would advise that you do this check in your model policy though

https://filamentphp.com/docs/2.x/admin/resources/getting-started#authorization

**Reply by eaziweb95** [reply]
When in put this code 
            ->actions([
                Action::make('delete')->hidden(fn ($record) => ! $record->actif)
            ])
a link delete appears but no action behind it and the edit action button has disappeared, how can we recreate actions buttons in this part of code ?

**Reply by danharrin** [reply]
Just add the check to the `delete()` method of your model policy instead

**Reply by eaziweb95** [reply]
Thanks a lot, it works !
